library/control/src/modules/WorkflowManager.rb

```diff
@@ -1427,6 +1489,93 @@ def DumpCurrentSettings
     publish function: :SetAllUsedControlFiles, type: "void (list <string>)"
     publish function: :HaveAdditionalWorkflows, type: "boolean ()"
     publish function: :DumpCurrentSettings, type: "map <string, any> ()"
+
+  private
+
+    # Find the product from a repository.
+    # @param repo_id [Fixnum] repository ID
+    # @return [Hash,nil] pkg-bindings product hash or nil if not found
+    def find_product(repo_id)
+      # identify the product
+      products = Pkg.ResolvableDependencies("", :product, "")
+      return nil unless products
+
+      products.select! { |p| p["source"] == repo_id }
+
+      if products.size > 1
+        log.warn("More than one product found in the repository: #{products}")
+        log.warn("Using the first one: #{products.first}")
+      end
+
+      products.first
+    end
+
+    # Find the extension package name for the specified release package.
+    # The extension package is defined by the "installerextension()"
+    # RPM "Provides" dependency.
+    # @return [String,nil] a package name or nil if not found
+    def find_control_package(release_package)
+      return nil unless release_package && release_package["deps"]
+
+      release_package["deps"].each do |dep|
+        provide = dep["provides"]
+        next unless provide && provide.match(/\Ainstallerextension\((.+)\)\z/)
+
+        control_file_package = Regexp.last_match[1].strip
```

Comment:
well, this is not so readable, to check where it is in fact defined.

What about 
```
next unless provide
control_file_package = provide[/\Ainstallerextension\((.+)\)\z/, 1]
next unless control_file_package
return control_file_package.strip
```